tools/wake-unit/unit.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "unit.h"
+#include "util/colour.h"
+#include "util/diagnostic.h"
+
+#include <set>
+#include <cstring>
+
+DiagnosticReporter *reporter;
+
+struct Test {
+  const char* test_name;
+  TestFunc test;
+  Test(const char* test_name_, TestFunc test_) : test_name(test_name_), test(test_) {}
+};
+
+// TODO: sequester this off to its own file and make it static
+// so that only TestRegister and the test runner can access it.
+std::vector<Test> tests__;
+
+TestRegister::TestRegister(const char* test_name, TestFunc test) {
+  tests__.emplace_back(test_name, test);
+}
+
+int main(int argc, char** argv) {
+  bool no_color = false;
+  if (argc > 2) {
+    std::cerr << "Too many arguments to be valid, --no-color is the only valid argument";
+    return -1;
```

Comment:
I try not to return negative numbers from main.

Reply:
Oh yeah. I forgot that 1 is the standard error return code, not sure why I though it was -1 for some reason. I think negative isn't specified by posix